src/main/java/org/mqttbee/mqtt/handler/connect/MqttConnectHandler.java

```diff
@@ -82,9 +82,20 @@ public MqttConnectHandler(
 
     @Override
     public void channelActive(final ChannelHandlerContext ctx) {
-        addClientData(ctx.channel());
-        writeConnect(ctx);
-        ctx.fireChannelActive();
+        if (!clientData.usesWebSockets()) {
+          addClientData(ctx.channel());
+          writeConnect(ctx);
+        }
+      ctx.fireChannelActive();
+    }
+
+   @Override
+    public void handlerAdded(ChannelHandlerContext ctx) {
+       super.handlerAdded(ctx);
+       if (clientData.usesWebSockets()) {
```

Comment:
Same here: mixing WebSockets with the ConnectHandler
Equivalent, but clearer is: if (channel.isActive())